vtpm/src/char.rs

```diff
@@ -0,0 +1,267 @@
+// Copyright © 2022, Microsoft Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+//
+
+use anyhow::anyhow;
+use nix::sys::socket::{recv, recvfrom, sendmsg, ControlMessage, MsgFlags};
+use nix::sys::uio::IoVec;
+use std::io::prelude::*;
+use std::io::Write;
+use std::os::unix::io::{AsRawFd, RawFd};
+use std::os::unix::net::UnixStream;
+use std::str;
+//use std::sync::{Arc, Mutex};
+use std::thread::sleep;
+use std::time::{Duration, Instant};
+use thiserror::Error;
+
+const TPM_TIS_BUFFER_MAX: usize = 4096;
```

Comment:
This should be moved out to a different place because this module seems to be for a specific implementation of the TPM backend.

The definition of the size of the buffer is not tied to that.